### PR TITLE
ExternalLocking4Rep : add registrar

### PR DESCRIPTION
**Pull request by orenyodfat**
- rename lock  to claim .

- claiming can be done for behalf of someone else only if it is register .

- add register function.

### CODE REVIEW

File: contracts/schemes/ExternalLocking4Reputation.sol [hunk 1]
```diff
@@ -10,20 +10,24 @@ import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 
 contract ExternalLocking4Reputation is Locking4Reputation, Ownable {
 
+    event Register(address indexed _beneficiary);
+
     address public externalLockingContract;
     string public getBalanceFuncSignature;
 
     // locker -> bool
     mapping(address => bool) public externalLockers;
```
Comment (dkent600) [inline contracts/schemes/ExternalLocking4Reputation.sol]: I think the code would be more clear if this were called "claimers".

File: contracts/schemes/ExternalLocking4Reputation.sol [hunk 2]
```diff
@@ -33,36 +37,45 @@ contract ExternalLocking4Reputation is Locking4Reputation, Ownable {
...
     /**
-     * @dev lock function
-     * @return lockingId
+     * @dev claim function
+     * @param _beneficiary the beneficiary address to claim for
+     *        if _beneficiary == 0 the claim will be for the msg.sender.
+     * @return claimId
      */
-    function lock() public returns(bytes32) {
+    function claim(address _beneficiary) public returns(bytes32) {
```
Comment (dkent600) [inline contracts/schemes/ExternalLocking4Reputation.sol]: Can we call "_beneficiary"  "_claimer"?

Comment (leviadam) [reply contracts/schemes/ExternalLocking4Reputation.sol]: But it is not the claimer, as one can claim for someone else, who was registered.

File: contracts/schemes/ExternalLocking4Reputation.sol (hunk 1 above)
Comment (dkent600) [reply contracts/schemes/ExternalLocking4Reputation.sol]: Even better:  "claimants".

File: contracts/schemes/ExternalLocking4Reputation.sol (hunk 2 above)
Comment (dkent600) [reply contracts/schemes/ExternalLocking4Reputation.sol]: @leviadam 

msg.sender in this case is claiming on behalf of the claimant (here `_beneficiary`), who "registered"  (poor word IMHO) to allow someone else (here msg.sender) to "claim" on their behalf. 

`msg.sender` is not here claiming anything, rather is executing a claim an behalf of someone else.

**Pull request merged**